### PR TITLE
Json sidecar nodatafile error

### LINKED ISSUE #358: Raise an error if a JSON sidecar is found with no corresponding data file
For example:
image
https://openneuro.org/datasets/ds001110/versions/00001

### PR DESCRIPTION
**Pull request by bGrass**
Fixes #358

### CODE REVIEW

**Comment by chrisgorgo** [comment]
That is a lot of files being modified... perhaps something went wrong when branching?

**Comment by bGrass** [comment]
Most of the modified files are a test dataset that I added in order to test the validation of this condition.

File: utils/files.js [hunk 1]
```diff
@@ -27,9 +27,24 @@ var fileUtils = {
...
   getBFileContent: getBFileContent,
+  dataExtRE: dataExtRE,
 }
 
 // implementations ----------------------------------------------------------------
+/**
+ * Make RegExp for detecting modalities from data file extensions
+ */
+function dataExtRE() {
+  return new RegExp(
```
Comment (nellh) [inline utils/files.js:38]: The RegExp here could just be exported, it doesn't need to be a function.

**Review by nellh — APPROVED** [review]
Made a comment about the regexp export but that's changing in another issue anyways, so I think this is good to go.

**Pull request merged**